server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -924,6 +1015,135 @@ private void mergeMetadataRecords(MergeInfo info) throws AccumuloException {
     }
   }
 
+  private boolean needsFencingForDeletion(MergeInfo info, KeyExtent keyExtent) {
+    return needsFencingForSplit(info, keyExtent) || info.deleteOverlaps(keyExtent);
+  }
+
+  // Instead of splitting or chopping tablets for a delete we instead create ranges
+  // to exclude the portion of the tablet that should be deleted
+  private List<Range> createRangesForDeletion(TabletMetadata tabletMetadata,
+      final KeyExtent deleteRange) {
+    final KeyExtent keyExtent = tabletMetadata.getExtent();
+    final Text prevTabletEndRow =
+        tabletMetadata.getPrevEndRow() != null ? tabletMetadata.getPrevEndRow() : null;
+    final Text tabletEndRow = keyExtent.endRow();
+    final Range tabletRange = tabletMetadata.getExtent().toDataRange();
+    final Key firstKey =
+        prevTabletEndRow != null ? new Key(prevTabletEndRow).followingKey(PartialKey.ROW) : null;
+    final List<Range> ranges = new ArrayList<>();
+
+    // This covers the case of when a deletion range overlaps the first tablet
+    // We need to create a range that excludes the deletion
+    if ((firstKey == null || !firstKey.equals(deleteRange.toDataRange().getStartKey()))
+        && deleteRange.toDataRange().getStartKey() != null
+        && tabletRange.contains(deleteRange.toDataRange().getStartKey())) {
+      Manager.log.trace(
+          "Fencing tablet at start of deletion range with end row {}, Start deletion row {}",
+          prevTabletEndRow, deleteRange.toDataRange().getStartKey().getRow());
+      ranges.add(new Range(prevTabletEndRow, false,
+          deleteRange.toDataRange().getStartKey().getRow(), false));
+    }
```

Comment:
Assuming that for the case where the first row of the delete range is null that it means everything in the tablet would be deleted from the perspective of the prev row.  If that is true then would not want to act when firstKey is null and could simplify the code to the following.  Not completely about this, also I did not adjust the log message.

```suggestion
    if (firstKey != null && tabletMetadata.getExtent().contains(firstKey.getRow())) {
      // In the case where firstKey is null, there is no data to keep in the tablet.
      Manager.log.trace(
          "Fencing tablet at start of deletion range with end row {}, Start deletion row {}",
          prevTabletEndRow, deleteRange.toDataRange().getStartKey().getRow());
      ranges.add(new Range(prevTabletEndRow, false, deleteRange.prevEndRow(), true));
    }
```

Reply:
I didn't test this but I'm assuming the changes here likely won't work for the same reasons as commented here: https://github.com/apache/accumulo/pull/3728#discussion_r1313531334